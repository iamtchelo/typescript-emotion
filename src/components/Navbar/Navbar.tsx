import * as React from 'react';
import { NavbarTitle, NavbarWrapper } from './navbar.elements';

export interface NavbarProps {
  title: string;
}

const Navbar: React.SFC<NavbarProps> = ({ title }) => (
  <NavbarWrapper>
    <NavbarTitle>{title}</NavbarTitle>
  </NavbarWrapper>
);

export default Navbar;