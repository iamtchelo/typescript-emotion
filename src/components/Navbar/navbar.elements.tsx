import styled from 'react-emotion';

export const NavbarWrapper = styled('nav')`
  padding: 15px 0;
  background: #4E3C88;
`;

export const NavbarTitle = styled('h3')`
  color: white;
  text-align: center;
`;