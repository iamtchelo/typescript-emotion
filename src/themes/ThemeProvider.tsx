import { ThemeProvider as Provider } from 'emotion-theming';
import * as React from 'react';
import { ComponentEnhancer, compose, setDisplayName ,withState } from 'recompose';
import primaryTheme from './primary';
import secondaryTheme from './secondary';

const themes = {
  primary: primaryTheme,
  secondary: secondaryTheme
};

export interface ThemeProviderProps {
  theme?: 'primary' | 'secondary';
  children?: React.SFC<any>;
};

export interface ThemeProviderDispatches {
  setTheme: (theme: 'primary' | 'secondary') => void;
};

export type EnhancedThemeProviderProps = ThemeProviderProps & ThemeProviderDispatches;

const ThemeProvider: React.SFC<EnhancedThemeProviderProps> = ({
  theme = 'primary',
  setTheme,
  children
}) => (
  <Provider theme={themes[theme]}>
    {children && children({ theme, setTheme })}
  </Provider>
);

const withTheme = compose(
  withState('theme', 'setTheme', ({ theme = 'primary' }: any) => theme)
);

const enhance = compose(
  setDisplayName('ThemeProvider'),
  withTheme
) as ComponentEnhancer<EnhancedThemeProviderProps, ThemeProviderProps>

export default enhance(ThemeProvider);
