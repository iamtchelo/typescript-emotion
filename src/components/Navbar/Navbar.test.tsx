import { render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import * as React from 'react';
import ThemeProvider from '../../themes';
import Navbar from './Navbar';

describe('<Navbar />', () => {
  test('should render a default navbar with a title', () => {
    const tree = render(
      <ThemeProvider>
        <Navbar title="My navbar" />
      </ThemeProvider>
    );
    expect(renderToJson(tree)).toMatchSnapshot();
  });
});