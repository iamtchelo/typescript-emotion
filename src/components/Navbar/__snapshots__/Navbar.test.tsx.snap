// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render a default navbar with a title 1`] = `
<nav
  class="css-14xwee0"
>
  <h3
    class="css-yzo0ge"
  >
    My navbar
  </h3>
</nav>
`;
